File: hsg9-tinh/2023-2024-thanhhoa/gacho/gacho.py
import os
import sys

input_file = os.path.join(sys.path[0], 'gacho.inp')
output_file = os.path.join(sys.path[0], 'gacho.out')

n = result = 0


def input():
    global n

    with open(input_file, 'r') as f:
        n = int(f.readline())


def process():
    global n, result

    k = (n - 1) // 2
    
    result = k * (n - k - 1)


def output():
    global result

    with open(output_file, 'w') as f:
        f.write(str(result))


if __name__ == '__main__':
    input()
    process()
    output()

File: hsg9-tinh/2023-2024-thanhhoa/gacho/gacho.cpp
#include <iostream>

#define input_file "gacho.inp"
#define output_file "gacho.out"

using namespace std;

typedef long long int lli;

lli n;
lli result;


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    freopen(input_file, "r", stdin);

    cin >> n;
}


void process()
{
    lli k = (n - 1) / 2;

    result = k * (n - k - 1);
}


void output()
{
    freopen(output_file, "w", stdout);
    cout << result;
}

int main()
{
    input();
    process();
    output();

    return 0;
}